src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/StringJoinFunction.java

```diff
@@ -0,0 +1,40 @@
+package sparksoniq.jsoniq.runtime.iterator.functions;
+
+import sparksoniq.exceptions.IteratorFlowException;
+import sparksoniq.exceptions.SparksoniqRuntimeException;
+import sparksoniq.exceptions.UnexpectedTypeException;
+import sparksoniq.jsoniq.item.Item;
+import sparksoniq.jsoniq.item.StringItem;
+import sparksoniq.jsoniq.item.metadata.ItemMetadata;
+import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
+import sparksoniq.jsoniq.runtime.iterator.functions.base.LocalFunctionCallIterator;
+import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
+
+import java.util.List;
+
+public class StringJoinFunction extends LocalFunctionCallIterator {
+    public StringJoinFunction(List<RuntimeIterator> arguments, IteratorMetadata iteratorMetadata) {
+        super(arguments, iteratorMetadata);
+        if (arguments.size() != 2)
+            throw new SparksoniqRuntimeException("Incorrect number of arguments for string-join function; " +
+                    "Exactly 2 arguments are required.");
```

Comment:
string-join() also exists with only one parameter. Then the second parameter is implicitly the empty string. That's why a function catalogue by name and arity will help here -- however of course, you can still reuse the same iterator for both string-join#1 and string-join#2.